logstash-core/spec/logstash/inputs/metrics_spec.rb

```diff
@@ -0,0 +1,32 @@
+# encoding: utf-8
+require "logstash/inputs/metrics"
+require "spec_helper"
+
+describe LogStash::Inputs::Metrics do
+  let(:queue) { [] }
+
+  describe "#run" do
+    it "should register itself to the collector observer" do
+      expect_any_instance_of(LogStash::Instrument::Collector).to receive(:add_observer).with(subject)
```

Comment:
I know this is the easy way to do it, and works, but could we figure it out a way to not use ```expect_any_instance_of```? the usage of this method is highly discouraged for the same rspec developers.